frontend/src/static/js/v_summary.js

```diff
@@ -141,19 +143,29 @@ window.vSummary = {
     },
     tmpFilterSinceDate() {
       if (this.tmpFilterSinceDate && this.tmpFilterSinceDate >= this.minDate) {
+        if (this.tmpFilterSinceDate !== this.minDate) { // If user modifies the since date field
```

Comment:
Should remove this guard clause. 
Reason: what if my intention was to share the link that was meant to use the min or max date?

Reply:
With the current implementation, if the user wants to share the link with min or max date, the user would have to modify date controls respectively and followed by setting the date controls back to the min or max date.

@damithc from our last discussion, we did not take into account when the user wants to share the link that was meant to use the min or max date. Your input?

Reply:
If the report was generated by specifying a start date, it should be captured in the URL when the report is opened the first time. Similarly, if there was an end date to begin with, it should be in the initial URL. i.e., the initial URL should match the dates specifically provided by the user via CLI.  Is that possible?

Reply:
@damithc yes it is possible.

Reply:
@damithc by implementing based on your comment, do we still want to remove the since date and until date hashes when the user resets the date range?

Reply:
@eugenepeh I have updated the code such that if the user did specify `sinceDate` and `untilDate` in the CLI, then `sinceDate` and `untilDate` hashes will be captured in the initial URL. On the other hand, if user did not specify `sinceDate` and `untilDate` in the URL, the hashes will not be captured in the initial URL.

So now if the user's intention was to share the link that uses min or max date, the user should specify the `sinceDate` and `untilDate` in the CLI arguments.

Reply:
Perhaps we can do this:
* The initial URL will contain dates matching CLI parameters e.g., if end date is not provided via CLI, the URL will not have an end date either.
* If a date is changed subsequently, the URL will take the new date e.g., if the start date is changed, the URL will change the start date too
* If a date is reset, the URL will take the reset date.

That means the user should not change dates if she wants to retain the behavior of the CLI parameters in the URL. If she resets dates, she will be sharing the report for the current period shown (i.e., not dynamic). I think this approach caters for all possibilities. What do you think?

Reply:
> What do you think?

Sounds good to me.

Reply:
Updated according to the requirements.

Reply:
Hmm, I still think that any modification to the date (even if going back to min or max date) should be considered modified. Because thinking at it logically, no one will go back to that specific date (without using reset), unless they noticed something is wrong, right?

Reply:
@eugenepeh yes, currently any modification to the date will be considered as modified. (Even if going back to min or max date)

Reply:
Hmm I see, because you're using watcher.

I kind of dislike this double "loading" style, because any error in coding will result in infinite loop. 
Can you try removing line 152 and modify 364 to ignore `!this.tmpFilterUntilDate` if it helps?

Reply:
Line 152 `this.tmpFilterSinceDate = this.filterSinceDate;` is dedicated for when the user clears the since date field using the HTML5 date picker `clear` button:

<img width="132" alt="Screen Shot 2019-07-28 at 6 04 28 PM" src="https://user-images.githubusercontent.com/32864116/62005232-ce720100-b162-11e9-9de0-94738a83d00c.png">

So if we remove line 152, the since date won't reset to the min date.

Reply:
I guess that is due to datepicker is doing `v-model="tmpFilterUntilDate"`.
What if we split the `v-model` 2 way binding to something like this?
```
   v-bind:value="something"
   v-on:input="something = $event.target.value"
```
So that the input variable and display variable are now different. Will we able to get rid of the double load this way?

Reply:
> I guess that is due to datepicker is doing `v-model="tmpFilterUntilDate"`.
> What if we split the `v-model` 2 way binding to something like this?

I have tried this, the dates still will not reset to their values when clicking the `clear` button.

Reply:
display wise or value wise? Note that the quote I gave above is just an example.
The actual value for  `v-bind:value` should be `filterUntilDate` itself and 
`v-on:input` should probably be `v-on:input="tmpFilterUntilDate = $event.target.value"`

Reply:
Actually my two-way binding implementation was something like:
`v-bind:value=filterSinceDate`, `v-on:input="updateTmpFilterSinceDate` 
where `updateTmpFilterSinceDate` sets `tmpFilterSinceDate` to `minDate` if `$event.target.value` is empty (which means user has cleared the since date), else sets `tmpFilterSinceDate` as `$event.target.value`.

Actually both `tmpFilterSinceDate` and `filterSinceDate` will reset to the `minDate` value when user has clicked the `clear` button, but I'm not sure why the UI still displays `dd/MM/yyyy` (cleared input) instead of the `minDate` value, .

Reply:
Realised what was the issue. I forgot to set `$event.target.value` back to `filterSinceDate` after clearing the since date field.

Removed the double loading style of tmpFilterSinceDate and tmpFilterUntilDate watcher.

Reply:
>where updateTmpFilterSinceDate sets tmpFilterSinceDate to minDate if $event.target.value is empty

Why do we need to set the `tmpFilterSinceDate` to minDate when its just a receiving variable though? The `tmpFilterSinceDate` watcher should naturally update the `filterSinceDate` isn't it?